src/main/java/frc/robot/maps/subsystems/SwerveDriveMap.java

```diff
@@ -0,0 +1,289 @@
+                // Max rotation (rad/s)
+                Math.PI,
+                // Gyro
+                new MockGyro());
+    }
+
+    public void updateInputs(Data io) {
+        io.frontLeftDrivePositionMeters = this.frontLeft.getDistance();
+        io.frontLeftVelocityMetersPerSec = this.frontLeft.getState().speedMetersPerSecond;
+        io.frontLeftPodAngle = this.frontLeft.getAngle().getDegrees();
+        io.frontLeftDriveCurrentAmps = this.frontLeft.getDriveMotor().getCurrentAmps();
+        io.frontLeftDriveTempC = this.frontLeft.getDriveMotor().getTemperatureC();
+        io.frontLeftSteeringCurrentAmps = this.frontLeft.getSteeringMotor().getCurrentAmps();
+        io.frontLeftSteeringTempC = this.frontLeft.getSteeringMotor().getTemperatureC();
+        this.frontLeft.setDesiredState(io.frontLeftDesired);
+
+        io.frontRightDrivePositionMeters = this.frontRight.getDistance();
+        io.frontRightVelocityMetersPerSec = this.frontRight.getState().speedMetersPerSecond;
+        io.frontRightPodAngle = this.frontRight.getAngle().getDegrees();
+        io.frontRightDriveCurrentAmps = this.frontRight.getDriveMotor().getCurrentAmps();
+        io.frontRightDriveTempC = this.frontRight.getDriveMotor().getTemperatureC();
+        io.frontRightSteeringCurrentAmps = this.frontRight.getSteeringMotor().getCurrentAmps();
+        io.frontRightSteeringTempC = this.frontRight.getSteeringMotor().getTemperatureC();
+        this.frontRight.setDesiredState(io.frontRightDesired);
+
+        io.rearLeftDrivePositionMeters = this.rearLeft.getDistance();
+        io.rearLeftVelocityMetersPerSec = this.rearLeft.getState().speedMetersPerSecond;
+        io.rearLeftPodAngle = this.rearLeft.getAngle().getDegrees();
+        io.rearLeftDriveCurrentAmps = this.rearLeft.getDriveMotor().getCurrentAmps();
+        io.rearLeftDriveTempC = this.rearLeft.getDriveMotor().getTemperatureC();
+        io.rearLeftSteeringCurrentAmps = this.rearLeft.getSteeringMotor().getCurrentAmps();
+        io.rearLeftSteeringTempC = this.rearLeft.getSteeringMotor().getTemperatureC();
+        this.rearLeft.setDesiredState(io.rearLeftDesired);
+
+        io.rearRightDrivePositionMeters = this.rearRight.getDistance();
+        io.rearRightVelocityMetersPerSec = this.rearRight.getState().speedMetersPerSecond;
+        io.rearRightPodAngle = this.rearRight.getAngle().getDegrees();
+        io.rearRightDriveCurrentAmps = this.rearRight.getDriveMotor().getCurrentAmps();
+        io.rearRightDriveTempC = this.rearRight.getDriveMotor().getTemperatureC();
+        io.rearRightSteeringCurrentAmps = this.rearRight.getSteeringMotor().getCurrentAmps();
+        io.rearRightSteeringTempC = this.rearRight.getSteeringMotor().getTemperatureC();
+        this.rearRight.setDesiredState(io.rearRightDesired);
+
+        io.gyroYawPositionDegrees = this.gyro.getAngle();
+    }
+
+    public static class Data implements LoggableInputs {
+        public double frontLeftDrivePositionMeters;
```

Comment:
You know what would be cool here? [Subtables](https://github.com/Mechanical-Advantage/AdvantageKit/blob/main/junction/core/src/org/littletonrobotics/junction/LogTable.java#L80)